mcu/include/uart.h

```diff
@@ -0,0 +1,23 @@
+#ifndef UART_H
+#define UART_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define BUF_SIZE 7
```

Comment:
The buffer size is set based upon the input data format, which has 6 characters: `s0123\r`. Therefore the buffer size is 7 to accommodate the null character at the end of the char array.

This dependency between the UART and the parser could potentially be cleaned up. The UART itself doesn't need to be tightly coupled to the parser, but the buffer size is inherently coupled between the parser and the UART. I'm not sure what the best way to handle this coupling is. At the least, the buffer size / command length should only be defined once.